internal/charmstore/store.go

```diff
@@ -742,53 +742,65 @@ func (s *Store) UpdateBaseEntity(url *router.ResolvedURL, update bson.D) error {
 	return nil
 }
 
+var ErrPublishResourceMismatch = errgo.Newf("charm published with incorrect resources")
```

Comment:
We need to update errToResp to make this return a StatusBadRequest, and probably test that somewhere too.